opennms-doc/guide-development/src/asciidoc/text/graph/graph.adoc

```diff
@@ -0,0 +1,201 @@
+
+* `GenericGraphContainer`,
+* `GenericGraph`,
+* `GenericVertex` and
+* `GenericEdge`
+
+Each of the above have besides their designated functionality described in link:#ga-development-graph-service-api-terminology[Terminology] also properties assigned to them.
+Some properties are mandatory and determined by the element's type.
+Others are optionally and totally up to the entity providing a graph to the system.
+
+The following rules apply to the _Generic Graph Model_:
+
+- Each Graph container must be uniquely identified.
+  This identifier is called a `container id`
+- Each Graph must be uniquely identified overall graphs (meaning the namespace must be unique through the system).
+  This identifier is called a `namespace`.
+- Each Vertex and Edge must be uniquely identified by an identifier.
+  This identifier is called an `id`.
+- Each Vertex and Edge have the same `namespace` as the Graph they are part of.
+  This allows to uniquely identify vertices and edges throughout the system.
+- Each Graph, Vertex or Edge have additional properties to define their nature in more detail.
+  E.g. a Vertex may contain a `label`, `node` or `location` property.
+- Each Edge contains the source and target id of the Vertex it connects, therefore is ALWAYS directed.
+- Each edge may reference Vertices from a different Graph (`namespace`).
+  However at least one side must share the same `namespace` as the edge itself.
+  This is the "owning" side, thus the edge will be a member of that Graph.
+
+===== Identifier and Namespace
+
+In order to uniquely identify each element and each graph container within the system different strategies are applied.
+
+1. The container is uniquely identified by an `id` property.
+2. The graph container is uniquely identified by a `namespace` property.
+3. Each element within the graph is uniquely identified by a `namespace` and `id` property.
+The `id` property must not be unique in the system, but within the graph.
+
+If a graph defines an `id` property as well, this SHOULD but MUST not be identically to the `namespace` property.
+However internally the `id` property is not used but provided to be more compatible with _GraphML_.
+
+The above rules allow to determine the "owner" of any given element if only that element is present.
+For example, if looking at any _Vertex_ one can tell to which graph it belongs due to the `namespace` property.
+
+Refer to module `features/graph/api` in package `org.opennms.netmgt.graph.api.generic` for implementation details.
+
+
+==== Domain Graph Model
+
+A _Domain Graph Model_ is a model implementing a use case in its domain using the _Generic Graph Model_.
+This can be achieved by simply using the _Generic Graph Model_ and setting the according properties.
+However in more complex use cases it may be necessary to provide a more Java-friendly way.
+Therefore basic Domain Graph objects are provided to act as a view on the _Generic Graph Model_ to provide a more convenient implementation.
```

Comment:
To make it even more clear maybe add something like : "Domain objects don't hold information on their own but provide only access to information that can be deducted from the underlying Generic Model and it's properties"

Reply:
Good point.